modules/components/analytics/core/org.wso2.carbon.pc.analytics.core/src/main/java/org/wso2/carbon/pc/analytics/core/ProcessLevelMonitor.java

```diff
@@ -0,0 +1,501 @@
+/**
+ * Copyright 2005-2015 WSO2, Inc. (http://wso2.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.pc.analytics.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.pc.analytics.core.clients.AnalyticsRestClient;
+import org.wso2.carbon.pc.analytics.core.models.AggregateField;
+import org.wso2.carbon.pc.analytics.core.models.AggregateQuery;
+import org.wso2.carbon.pc.analytics.core.models.SearchQuery;
+import org.wso2.carbon.pc.analytics.core.utils.Helper;
+
+import java.util.ArrayList;
+import java.util.Hashtable;
+
+/**
+ * ProcessLevelMonitor holds all the functionalites for the process level monitoring
+ */
+public class ProcessLevelMonitor {
+	private static final Log log = LogFactory.getLog(ProcessLevelMonitor.class);
+
+	/**
+	 * perform query: SELECT processDefinitionId, AVG(duration) AS avgExecutionTime FROM
+	 * PROCESS_USAGE_SUMMARY WHERE <date range> GROUP BY processDefinitionId;
+	 *
+	 * @param filters is a given date range represents as the JSON string
+	 * @return the result as a JSON string
+	 */
+	public String getAvgExecuteTimeVsProcessId(String filters) {
```

Comment:
Code formatting: use 4 spaces for indentation instead of tab character (this can be set in your IDE options)